specs/protocol/messengers.md

```diff
@@ -79,6 +84,16 @@ then waits for the finalization window to pass, and then finalizes the withdrawa
 on the OptimismPortal, which calls `relayMessage` on the
 `L1CrossDomainMessenger` to finalize the withdrawal.
 
+## Rolling a Message Back
+
+The `sendHashToRollbackInbox` function is used to send the hashes of failed
+and unsuccesful messages to the `ROLLBACK_INBOX` contract on the other domain
+after a given delay, ensuring the message corresponding to the rolled back hash
+is never processed in the current domain. The main purpose of this function
+is to inform the other domain of failed messages so contracts living in
+this other domain can handle the message failure. This allows functionalities
```

Comment:
Do you see any other use cases besides unlocking stuck ERC20s? Trying to get a sense of what the major use cases built on this would be

Reply:
that's a great question, yes.
we think this will be a great primitive for handling the not-so-happy path of cross-chain actions.
another example would be a cross-chain(x) swap, where the origin.amm would forward and save an identifier of the user xAction.
which, if the xMessage fails, will be used to continue and gracefully handle the reverted xAction, returning the funds or an LP position to the swapper.